core/chains/evm/txmgr/stuck_tx_detector.go

```diff
@@ -217,16 +217,24 @@ func (d *stuckTxDetector) detectStuckTransactionsHeuristic(ctx context.Context,
 		}
 		// Tx attempts are loaded from newest to oldest
 		oldestBroadcastAttempt, newestBroadcastAttempt, broadcastedAttemptsCount := findBroadcastedAttempts(tx)
+		d.lggr.Debugf("found %d broadcasted attempts for tx id %d in stuck transaction heuristic", broadcastedAttemptsCount, tx.ID)
+
 		// 2. Check if Threshold amount of blocks have passed since the oldest attempt's broadcast block num
+		if oldestBroadcastAttempt == nil || oldestBroadcastAttempt.BroadcastBeforeBlockNum == nil {
+			d.lggr.Errorf("oldestBroadcastAttempt is nil or BroadcastBeforeBlockNum is nil for tx id %v in stuck transaction heuristic", tx.ID)
+			continue
+		}
```

Comment:
Actually think if this is a valid scenario, we should log this as debug to avoid too much noise. Also might be worth splitting this into 2 different conditions and logs so it's easier for us to follow what happened. We should also adjust the message so it's not explicitly referencing the code and easier for people to understand what happened
```suggestion
		if oldestBroadcastAttempt == nil {
			d.lggr.Debugw("failed to find broadcast attempt for tx in stuck transaction heuristic", "tx", tx)
			continue
		}
		if oldestBroadcastAttempt.BroadcastBeforeBlockNum == nil {
		        d.lggr.Debugw("BroadcastBeforeBlockNum not set for broadcast attempt in stuck transaction heuristic", "attempt", oldestBroadcastAttempt)
			continue
		}
```